zcncore/transaction_query_base.go

```diff
@@ -184,6 +188,117 @@ func (tq *TransactionQuery) randOne(ctx context.Context) (string, error) {
 	}
 }
 
+func (tq *TransactionQuery) FromConsensus(ctx context.Context, query string, handle QueryResultHandle) error {
+
+	min := getMinShardersVerify()
```

Comment:
The min could be either too small such as 0, or too large if the network is large, so we still need to check the boundary. So let's limit it to 3 <= min <= 10.

Reply:
```
func getMinShardersVerify() int {
	minSharders := util.MaxInt(calculateMinRequired(float64(_config.chain.MinConfirmation), float64(len(_config.chain.Sharders))/100), 1)
	logging.Info("Minimum sharders used for verify :", minSharders)
	return minSharders
}
```
the minimal value is 1. 10 should not work if there are 30 sharders, and `min_confirmation: 50`

Reply:
min should be 3 or all of sharders if it is less 3 , right?
how it works? should we also change it? 
```
// consensus has been reached, don't waiting for other requests
				if maxConsensus*100/tq.max >= consensusThresh {
					return true, true
				}
```

let me repeat your request
- send query requests to 3-10 sharders
- check maxConsensus * 100/10 >= consensusThresh, if yes, it is consensus result. 

right?

Reply:
you want the majority of your requests to return the same value most of the time. in general, you have some `trusted_threshold` for your sharders, and you want that

``requests_same_result/requests_sent > trusted_threshold``

imagine `trusted_threshold=0.4` and if you made 7 requests and 1 sharder returned success and 4 returned error and 2 didn't respond, you should check for the dominant one (4 errors) with formula

``4/7 > 0.4`` so you get the consensus on error response